main/define/src/mill/define/Discover.scala

```diff
@@ -16,15 +16,24 @@ import scala.reflect.macros.blackbox
  * the `T.command` methods we find. This mapping from `Class[_]` to `MainData`
  * can then be used later to look up the `MainData` for any module.
  */
-case class Discover[T] private (val value: Map[
-  Class[_],
-  (Seq[String], Seq[mainargs.MainData[_, _]])
-])
+case class Discover[T] private (
+    val value: Map[
+      Class[_],
+      (Seq[String], Seq[mainargs.MainData[_, _]])
+    ],
+    dummy: Int = 0 /* avoid conflict with Discover.apply(value: Map) below*/
+) {
+  def this(value: Map[Class[_], Seq[mainargs.MainData[_, _]]]) =
+    this(value.mapValues((Nil, _)).toMap)
```

Comment:
This isn't cross compiled or anything right? You can use `.view` here then to get rid of the deprecation warning:

```suggestion
    this(value.view.mapValues((Nil, _)).toMap)
```